### PR TITLE
Add .NET Core support to DLR

### PR DESCRIPTION
**Pull request by slozier**

### CODE REVIEW

File: Build/Common.proj [hunk 1]
```diff
@@ -59,8 +59,7 @@
 
   <PropertyGroup Condition="'$(ReferencedPlatform)' == ''">
     <ReferencedPlatform>net45</ReferencedPlatform>
```
Comment (slide) [inline Build/Common.proj]: Do we need ReferencedPlatform at all anymore?

File: Build/Common.proj [hunk 2]
```diff
@@ -73,7 +72,7 @@
   </PropertyGroup>
 
   <!-- References -->
-  <Import Project="$(DlrBuildDir)\$(ReferencedPlatform).props" />
```
Comment (slide) [inline Build/Common.proj]: Is TargetFramework always set? If not, don't we need this so that it pulls in when it's not set?

File: Dlr.sln [hunk 3]
```diff
@@ -53,10 +52,6 @@ Global
 		Release|Any CPU = Release|Any CPU
 	EndGlobalSection
 	GlobalSection(ProjectConfigurationPlatforms) = postSolution
-		{3272E46C-F4D9-418D-9EE9-081A1B658EE4}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
```
Comment (slide) [inline Dlr.sln]: What project was removed?

File: Src/Microsoft.Dynamic/Generation/FieldBuilderExpression.cs [hunk 4]
```diff
@@ -31,9 +31,19 @@ namespace Microsoft.Scripting.Generation {
     public class FieldBuilderExpression : Expression {
         private readonly FieldBuilder _builder;
 
+#if NETCOREAPP2_0
+        private readonly StrongBox<Type> _finishedType;
```
Comment (slide) [inline Src/Microsoft.Dynamic/Generation/FieldBuilderExpression.cs]: Can we do this as a runtime check instead, e.g. make finishedType default to null and then check if its null in GetFieldInfo?

File: Build/Common.proj (hunk 1 above)
Comment (slozier) [inline Build/Common.proj]: I was planning on eventually getting rid of it, but it's somewhat unrelated to the .NET Core support so I was going to do it as a separate PR.

Comment (slide) [inline Build/Common.proj]: Ok, as long as it is on the radar.

File: Build/Common.proj (hunk 2 above)
Comment (slozier) [inline Build/Common.proj]: With the new csproj files TargetFramework should always be set. I'm not exactly sure how it behaves when you mix with old style csproj files...

File: Dlr.sln (hunk 3 above)
Comment (slozier) [inline Dlr.sln]: Microsoft.Scripting.AspNet. I don't mind re-adding it, but it should be updated to the new csproj format if we do.

Comment (slide) [inline Dlr.sln]: Removing that one is fine

File: Src/Microsoft.Dynamic/Generation/FieldBuilderExpression.cs (hunk 4 above)
Comment (slozier) [inline Src/Microsoft.Dynamic/Generation/FieldBuilderExpression.cs]: The code wouldn't compile since there's no FieldBuilder.GetToken in .NET Core. I just readded the code that was there before it was removed in the Silverlight cleanup.

Comment (slide) [inline Src/Microsoft.Dynamic/Generation/FieldBuilderExpression.cs]: :+1:

File: Build/Common.proj (hunk 2 above)
Comment (slide) [inline Build/Common.proj]: Why the need for the Condition in the import of $(TargetFramework).props then?

Comment (slozier) [inline Build/Common.proj]: I'm not super familiar with how it works, but msbuild does an "outer" build and an "inner" build. In the "outer" build pass TargetFramework is not set. You can try to read about it if you want, I don't understand much of it: https://oren.codes/2017/01/04/multi-targeting-the-world-a-single-project-to-rule-them-all/#how-it-works

File: Src/Microsoft.Dynamic/Microsoft.Dynamic.csproj [hunk 5]
```diff
@@ -1,8 +1,8 @@
-﻿<Project>
-  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (gpetrou) [inline Src/Microsoft.Dynamic/Microsoft.Dynamic.csproj]: Is there a reason that this changed? Did you find a problem with the current implementation?

File: Tests/ClrAssembly/ClrAssembly.csproj [hunk 6]
```diff
@@ -74,7 +75,11 @@
   </Target>
   <Target Name="CSManyArea" Inputs="@(AreaCSFile)" Outputs="@(AreaCSFile->'$(OutputPath)\$(Prefix)%(Filename).dll')" DependsOnTargets="TypeSample">
     <Csc TargetType="library" Sources="Src\fieldtests.cs" OutputAssembly="$(OutputPath)\$(Prefix)fieldtests.dll" References="$(OutputPath)\$(Prefix)typesamples.dll" />
-    <Csc TargetType="library" Sources="Src\methodargs.cs" OutputAssembly="$(OutputPath)\$(Prefix)methodargs.dll" References="System.dll;$(OutputPath)\Microsoft.Scripting.dll;$(OutputPath)\$(Prefix)typesamples.dll" />
+    <PropertyGroup>
+      <SystemRuntimeDll></SystemRuntimeDll>
```
Comment (gpetrou) [inline Tests/ClrAssembly/ClrAssembly.csproj]: Is this needed?

Comment (slozier) [inline Tests/ClrAssembly/ClrAssembly.csproj]: You mean the empty assignment? Not sure, will try.

File: Src/Microsoft.Dynamic/Microsoft.Dynamic.csproj (hunk 5 above)
Comment (slozier) [inline Src/Microsoft.Dynamic/Microsoft.Dynamic.csproj]: Yes, the current implementation was causing issues (although at this point I don't remember what exactly).
